sklearn/datasets/_samples_generator.py

```diff
@@ -621,8 +624,16 @@ def make_circles(n_samples=100, shuffle=True, noise=None, random_state=None,
     if factor >= 1 or factor < 0:
         raise ValueError("'factor' has to be between 0 and 1.")
 
-    n_samples_out = n_samples // 2
-    n_samples_in = n_samples - n_samples_out
+    if isinstance(n_samples, int):
```

Comment:
There might be problems with numpy types. I always forget what the state of the art way to check for both numpy and python ints is

Reply:
You mean `isinstance(n_samples, (int, np.integer))` mentioned [here](https://stackoverflow.com/questions/37726830/how-to-determine-if-a-number-is-any-type-of-int-core-or-numpy-signed-or-not)?